Remove response inline script

poi33 opened the pull request:
Added a service to that gets the ga script

alansemenov reviewed (changes requested):
That's too many attributes. We already have a practice of doing it via config service, we should use the same here. Check any system app or Content Studio or recent Siteimprove or VWO changes. The only attribute we need to pass to the script is URL of the config service.

Use [fetch API](https://developer.mozilla.org/en-US/docs/Web/API/fetch)to fetch data from the service.

Also, id of the widget element should always be "`widget-<appId>`", as it was before.

alansemenov reviewed (changes requested):
Why not make it consistent with other apps? We don't want to do it in X different ways just because there exist X different ways of doing this.

1. Generate config service url **with contentId** in URL, like [here](https://github.com/enonic/app-vwo/blob/master/src/main/resources/admin/widgets/vwo-campaign/vwo-campaign.js#L37). There's absolutely no point in sending one contentId parameter in body and doing it via POST.

2. Pass config service url via data attribute **directly to the GA script file**, like [here](https://github.com/enonic/app-vwo/blob/master/src/main/resources/admin/widgets/vwo-campaign/vwo-campaign.html#L3). We don't need any hidden divs. The only reason to not do that is support of legacy browsers which don't support `document.currentScript` but we don't even support them in Content Studio anymore.

3. Get config service url from `document.currentScript` like [here](https://github.com/enonic/app-vwo/blob/master/src/main/resources/assets/js/app-vwo.js#L812), add a check for non-empty value.

4. Change service to `GET` and get `contentId` from req.params.

5. We should use `clearInterval` if you switched to `setInterval`.

Nice idea with having gascript in a service (but use `const` instead of `let` for snippet)

poi33 commented:
> Why not make it consistent with other apps? 

The first attempt was improve to make it more consistent. 
I try to follow the style of each application. Eg. keep the coding style and general idea behind the structure. 
Most of our apps are widly different in how they are made. What kind of coding style was used and changed over time with other maintainers. 
But its unavoidable that every developer adds his own way into the code. 
I know my style of coding is very explicit. This is becuase for some reason everything needs to be as short as possible in modern javascript. 
My opinion is that this makes code hard to read and more difficult to maintain. If any of my code is functionally worse or does not work as well as another solution would please tell me so and I will improve it based on it. 

> 1. Generate config service url with contentId in URL, like [here](https://github.com/enonic/app-vwo/blob/master/src/main/resources/admin/widgets/vwo-campaign/vwo-campaign.js#L37). There's absolutely no point in sending one contentId parameter in body and doing it via POST.

To keep it consistent and since you prefer it changed.
The reason I don't like it:
I don't like that the param is never present in the google-analytics script. Its never shown or mentioned. 
From the generatio nof the param ga-report.js -> google-analytics.js -> gaconfig.js. the gaconfig is the first to access it, so its difficult for a developer to locate this value.

> 2. Pass config service url via data attribute directly to the GA script file, like [here](https://github.com/enonic/app-vwo/blob/master/src/main/resources/admin/widgets/vwo-campaign/vwo-campaign.html#L3). We don't need any hidden divs. The only reason to not do that is support of legacy browsers which don't support document.currentScript but we don't even support them in Content Studio anymore.

> To keep it consistent changed it.
I don't see much different between having an tag for the storage and a tag for the script and combining them. 
Having seperate elements makes it easier to find in the html structure.

> 3 Get config service url from document.currentScript like [here](https://github.com/enonic/app-vwo/blob/master/src/main/resources/assets/js/app-vwo.js#L812), add a check for non-empty value.

Not heard about document.currentScript before. Useful, changed to use it.

> 4. Change service to GET and get contentId from req.params.

Changed for consistency. I don't see any difference between a post body and a urlQuery param.

> 5. We should use clearInterval if you switched to setInterval.

Tried to find a way to make this work without waiting. Could not, and changed it back to setTimout.
